external-crates/move/move-compiler/src/typing/core.rs

```diff
@@ -982,7 +1009,11 @@ pub fn make_function_type(
     };
 
     let defined_loc = finfo.defined_loc;
+    let is_public_for_testing =
+        is_public_for_testing(context.env, context.current_package, f, finfo.entry);
+    let is_testing_context = context.is_testing_context();
     match finfo.visibility {
+        _ if is_testing_context && is_public_for_testing.is_some() => (),
```

Comment:
Would it make sense to still enforce some of the package/friend visibility here?

Reply:
I do not understand. The purpose is to override the visibility for entry functions for test contexts, there is nothing to check for package/friend?

Reply:
That makes sense, yeah.